.atomist/editors/RugGeneratorFunctions.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2017 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Project } from '@atomist/rug/model/Core'
+import { File } from '@atomist/rug/model/File'
+import { PathExpression, PathExpressionEngine } from '@atomist/rug/tree/PathExpression'
+
+export function cleanReadMe(project: Project, projectName: string, description: string, owner: string): void {
+    let eng: PathExpressionEngine = project.context().pathExpressionEngine();
+
+    let readMePE = new PathExpression<Project, File>("/*[@name='README.md']");
+    let readMe: File = eng.scalar(project, readMePE);
+    readMe.replace("# Atomist 'spring-boot-rest-service'", "# " + projectName);
+    readMe.regexpReplace("This Rug archive contains a generator for a Spring Boot Rest Service project.[\\s\\S]*?\n## Rugs\n", description + "\n");
```

Comment:
If you have the following text in the README:

```markdown
This [Rug][rug] project contains a generator for a [Spring Boot][boot]
[REST][rest] service project.

[rug]: http://docs.atomist.com/
[boot]: https://projects.spring.io/spring-boot/
[rest]: https://en.wikipedia.org/wiki/Representational_state_transfer

## Rugs
```

just alter the regex to skip the bad parts. Something like,

```typescript
readMe.regexpReplace("This .*?Rug.*? project contains a generator for a .*?Spring Boot[\\s\\S]*?\n## Rugs\n", description + "\n\n##Rugs\n");
```